pkg/util/lifted/lua/string_safe.go

```diff
@@ -0,0 +1,382 @@
+/*
+Copyright 2025 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lua
+
+// string_safe contains a subset of the lua String library. For security reasons, we do not expose
+// the entirety of lua String library to custom actions, such as string.gsub(), string.rep(), etc.
+// Only the safe functions like string.len() are exposed. In addition, strict limits have been placed
+// on the length of the processed string. Implementation was copied from
+// github.com/yuin/gopher-lua.
```

Comment:
We should explain the code by :
1. explain where it is copied from
2. what you modified and why